### PR TITLE
TST: parametrize and de-duplicate arith tests

### PR DESCRIPTION
**Pull request by jbrockmendel**
Hopefully before long we can get rid of `box` fixture altogether, then rename `box_with_array` to `box`

### CODE REVIEW

File: pandas/tests/arithmetic/conftest.py [hunk 1]
```diff
@@ -190,7 +190,12 @@ def box(request):
 
 
 @pytest.fixture(
-    params=[pd.Index, pd.Series, pytest.param(pd.DataFrame, marks=pytest.mark.xfail)],
+    params=[
+        pd.Index,
+        pd.Series,
+        pytest.param(pd.DataFrame, marks=pytest.mark.xfail),
```
Comment (gfyoung) [inline pandas/tests/arithmetic/conftest.py]: Is there documentation as to why we `XFAIL` this?

File: pandas/tests/arithmetic/test_datetime64.py [hunk 2]
```diff
@@ -599,15 +577,18 @@ def test_comparison_tzawareness_compat(self, op, box_df_fail):
         with pytest.raises(TypeError, match=msg):
             op(dz, np.array(list(dr), dtype=object))
 
-        # Check that there isn't a problem aware-aware and naive-naive do not
-        # raise
+        # Check that there isn't a problem aware-aware and naive-naive do not raise
```
Comment (gfyoung) [inline pandas/tests/arithmetic/test_datetime64.py]: I'm not sure what this comment means.  Might be a wording or grammar thing.

File: pandas/tests/arithmetic/conftest.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/arithmetic/conftest.py]: it's case-by-case in the places where it is used

File: pandas/tests/arithmetic/test_datetime64.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/arithmetic/test_datetime64.py]: will re-word, thanks

**Comment by jreback** [comment]
thanks @jbrockmendel

**Pull request merged**